### PR TITLE
Fix #2450: Fix test testRecentlyPlayedTestActivity_clickOnToolbarNavigationButton_closeActivity [Testing]

### LINKED ISSUE #2450: Fix test testRecentlyPlayedTestActivity_clickOnToolbarNavigationButton_closeActivity [Testing]
**Describe the bug**
The test `testRecentlyPlayedTestActivity_clickOnToolbarNavigationButton_closeActivity` is only performing action.
Find this test in `RecentlyPlayedFragmentTest` app module.

**Expected behavior**
3 components of Espresso test 
1. ViewMatchers
2. ViewActions
3. ViewAssertions

There are no ViewAssertions in this test case, which means this test is checking nothing. As per the test name, we need to test the activity is closing or not.

### PR DESCRIPTION
**Pull request by harsh-hash**
Fixes #2450 
i have added ViewAssertion in test testRecentlyPlayedTestActivity_clickOnToolbarNavigationButton_closeActivity using ActivityScenarioRule

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

File: app/src/sharedTest/java/org/oppia/android/app/home/RecentlyPlayedFragmentTest.kt [hunk 1]
```diff
@@ -155,6 +163,7 @@ class RecentlyPlayedFragmentTest {
     ).use {
       testCoroutineDispatchers.runCurrent()
       onView(withId(R.id.recently_played_toolbar)).perform(click())
+      assertThat(activityScenarioRule.scenario.state).isEqualTo(Lifecycle.State.DESTROYED)
```
Comment (anandwana001) [inline app/src/sharedTest/java/org/oppia/android/app/home/RecentlyPlayedFragmentTest.kt:166]: Can we do this using `activityScenarioRule`?

Use Reference from `testStoryFragment_clickOnToolbarNavigationButton_closeActivity`

**Review by anandwana001 — CHANGES_REQUESTED** [review]

File: app/src/sharedTest/java/org/oppia/android/app/home/RecentlyPlayedFragmentTest.kt (hunk 1 above)
Comment (harsh-hash) [reply app/src/sharedTest/java/org/oppia/android/app/home/RecentlyPlayedFragmentTest.kt:166]: @anandwana001 I don't find any method using activityScenarioRule which can ensure the activity is finished or not.

Comment (anandwana001) [reply app/src/sharedTest/java/org/oppia/android/app/home/RecentlyPlayedFragmentTest.kt:166]: Check this method if it helps as reference https://github.com/oppia/oppia-android/blob/eb137fba7718bcf5b4eea5c379d727e88166607f/app/src/sharedTest/java/org/oppia/android/app/story/StoryFragmentTest.kt#L131

Comment (harsh-hash) [reply app/src/sharedTest/java/org/oppia/android/app/home/RecentlyPlayedFragmentTest.kt:166]: I have also checked this. In this test we are using activityTestRule to check the activity is finished. using activityScenarioRule i am able to get the activity using activityScenarioRule.javaClass but not able to find any method which can ensure the activity is finished or not

**Comment by anandwana001** [comment]
@harsh-hash Could you check the test on Robolectric?

it's failing on GitHubActions

```
org.oppia.android.app.home.RecentlyPlayedFragmentTest > testRecentlyPlayedTestActivity_clickOnToolbarNavigationButton_closeActivity FAILED
    expected: DESTROYED
    but was : RESUMED
        at org.oppia.android.app.home.RecentlyPlayedFragmentTest.testRecentlyPlayedTestActivity_clickOnToolbarNavigationButton_closeActivity(RecentlyPlayedFragmentTest.kt:166)
```

Also, add the screenshot showing the test is passing on both Espresso and Robolectric. 
Check here if you want to see how to run test case on Robolectric -  https://github.com/oppia/oppia-android/wiki/Oppia-Android-Testing#running-robolectric-tests

**Comment by harsh-hash** [comment]
> @harsh-hash Could you check the test on Robolectric?
> 
> it's failing on GitHubActions
> 
> ```
> org.oppia.android.app.home.RecentlyPlayedFragmentTest > testRecentlyPlayedTestActivity_clickOnToolbarNavigationButton_closeActivity FAILED
>     expected: DESTROYED
>     but was : RESUMED
>         at org.oppia.android.app.home.RecentlyPlayedFragmentTest.testRecentlyPlayedTestActivity_clickOnToolbarNavigationButton_closeActivity(RecentlyPlayedFragmentTest.kt:166)
> ```
> 
> Also, add the screenshot showing the test is passing on both Espresso and Robolectric.
> Check here if you want to see how to run test case on Robolectric - https://github.com/oppia/oppia-android/wiki/Oppia-Android-Testing#running-robolectric-tests

@anandwana001 I hadn't checked this on roboelectric . yes its failing on roboelectric. can you give some suggestion on that? I have also tried it by ActivityScenario returned by the ActivityScenario.launch<> but again its failing.

**Comment by anandwana001** [comment]
Let's check this out and see if it helps - http://robolectric.org/androidx_test/

**Comment by anandwana001** [comment]
Any update on this @harsh-hash ?
Also, please resolve the conflicts.

File: app/src/sharedTest/java/org/oppia/android/app/home/RecentlyPlayedFragmentTest.kt (hunk 1 above)
Comment (Srikanth-Kb) [reply app/src/sharedTest/java/org/oppia/android/app/home/RecentlyPlayedFragmentTest.kt:166]: Using ActivityScenarioRule, I think this is the only way to obtain the state of an activity. 
https://github.com/android/android-test/issues/235

**Comment by anandwana001** [comment]
Closing this PR as no further update.

**Pull request closed without merging**